server/src/main/java/org/apache/iotdb/db/engine/upgrade/UpgradeTask.java

```diff
@@ -40,30 +47,69 @@ public UpgradeTask(TsFileResource upgradeResource) {
   @Override
   public void runMayThrow() {
     try {
-      upgradeResource.getWriteQueryLock().readLock().lock();
-      String tsfilePathBefore = upgradeResource.getFile().getAbsolutePath();
-      String tsfilePathAfter = UpgradeUtils.getUpgradeFileName(upgradeResource.getFile());
-
-      UpgradeLog.writeUpgradeLogFile(
-          tsfilePathBefore + COMMA_SEPERATOR + UpgradeCheckStatus.BEGIN_UPGRADE_FILE);
+      List<TsFileResource> upgradedResources = generateUpgradedFiles();
       upgradeResource.getWriteQueryLock().writeLock().lock();
+      String oldTsfilePath = upgradeResource.getFile().getAbsolutePath();
       try {
-        FSFactoryProducer.getFSFactory().getFile(tsfilePathBefore).delete();
-        FSFactoryProducer.getFSFactory()
-            .moveFile(FSFactoryProducer.getFSFactory().getFile(tsfilePathAfter),
-                FSFactoryProducer.getFSFactory().getFile(tsfilePathBefore));
+        // delete old TsFile
+        upgradeResource.remove();
+        // move upgraded TsFiles to their own partition directories
+        for (TsFileResource upgradedResource : upgradedResources) {
+          File upgradedFile = upgradedResource.getFile();
+          long partition = upgradedResource.getTimePartitionWithCheck();
```

Comment:
the upgradedResources is under control, no need to getTimepartition with check

Reply:
Fixed